### PR TITLE
[WIP] Plug leak

### LINKED ISSUE #1881: [prism-cli] Memory leak while validating
## Describe the bug

Following #1860 by @lukasz-kuzynski-11sigma we've been able to bump from Prism 4.1.2 to Prism 4.3.4.

However, some of servers started to restart due to Out of Memory issues. We've tracked down the issue to `validateInput` and `validateOutput` functions leaking memory.

## To Reproduce

A complete standalone repro case is available at https://github.com/nulltoken/prism_repro

It triggers a loop of 1000 request and response validations, capturing the amount of used Heap every 10 steps, and triggering a garbage collection between every step.

Running `yarn repro` shows the following output where one can clearly see the heap growing.

```
$ yarn repro
yarn run v1.22.5
$ yarn node -r ts-node/register --unhandled-rejections=strict --expose-gc ./src/repro.ts
unknown format "JSON Web Token RFC 7519" ignored in schema at path "#/properties/authorization"
unknown format "JSON Web Token RFC 7519" ignored in schema at path "#/properties/authorization"
[...snipped for brevity...]
unknown format "JSON Web Token RFC 7519" ignored in schema at path "#/properties/authorization"
unknown format "JSON Web Token RFC 7519" ignored in schema at path "#/properties/authorization"

i	HeapUsed
0	114059736
10	113304496
20	113369808
30	113711088
[...snipped for brevity...]
930	131688256
940	131704024
950	131884168
960	132243528
970	132365232
980	132556000
990	132876080

Running prism v4.3.4
Done in 15.77s.
```

## Expected behavior

No leak ? ;-)

/cc @chohmann

### PR DESCRIPTION
**Pull request by nulltoken**
Fixes #1881 

**Summary**

Replace this with something that explains what this PR is for and why it matters.

**Checklist**

- The basics
  - [ ] I tested these changes manually in my local or dev environment
- Tests
  - [ ] Added or updated
  - [ ] N/A
- Event Tracking
  - [ ] I added event tracking and followed the event tracking guidelines
  - [ ] N/A
- Error Reporting
  - [ ] I reported errors and followed the error reporting guidelines
  - [ ] N/A

**Screenshots**

If applicable, add screenshots or gifs to help demonstrate the changes. If not applicable, remove this screenshots
section before creating the PR.

**Additional context**

Add any other context about the pull request here. Remove this section if there is no additional context.

### CODE REVIEW

File: packages/cli/src/operations.ts [hunk 1]
```diff
@@ -3,22 +3,65 @@ import { transformOas2Operations } from '@stoplight/http-spec/oas2/operation';
...
 
-  let operations: IHttpOperation[] = [];
+  let operations: IHttpOperationEx[] = [];
   if (isOpenAPI2(result)) operations = transformOas2Operations(result);
   else if (isOpenAPI3(result)) operations = transformOas3Operations(result);
   else if (isPostmanCollection(result)) operations = transformPostmanCollectionOperations(result);
   else throw new Error('Unsupported document format');
 
   operations.forEach((op, i, ops) => {
+    if (op.request !== undefined) {
```
Comment (radzserg) [inline packages/cli/src/operations.ts:34]: Yeah I see your point, make sense

**Comment by nulltoken** [comment]
@radzserg Hmmm. I've been sidetracked by another topic and thus haven't had the time to fix all the tests. I've still got some work in that area.

However, even if the CI doesn't pass, you might be willing to take an early peek at this WIP.

Feedback would be very welcome ;-)

**Comment by nulltoken** [comment]
Superceded by #1892

**Comment by lukasz-kuzynski-11sigma** [comment]
Closing due to fix: #1892

**Pull request closed without merging**